url89cafed56a_AlexDiru_esper_compiler_final_tgz-pJ8-compiler_ProgramJ8/src/compiler/EsperParser.java

```diff
@@ -0,0 +1,1997 @@
+
+                    if ( input.LA(1)==MINUS||input.LA(1)==PLUS ) {
+                        input.consume();
+                        root_0 = (Object)adaptor.becomeRoot(
+                        (Object)adaptor.create(set11)
+                        , root_0);
+                        state.errorRecovery=false;
+                    }
+                    else {
+                        MismatchedSetException mse = new MismatchedSetException(null,input);
+                        throw mse;
+                    }
+
+
+                    pushFollow(FOLLOW_expr_in_expr200);
+                    expr12=expr();
+
+                    state._fsp--;
+
+                    adaptor.addChild(root_0, expr12.getTree());
+
+                    pushFollow(FOLLOW_expr_in_expr202);
+                    expr13=expr();
+
+                    state._fsp--;
+
+                    adaptor.addChild(root_0, expr13.getTree());
+
+                    }
+                    break;
+                case 2 :
+                    // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:32:11: term
+                    {
+                    root_0 = (Object)adaptor.nil();
+
+
+                    pushFollow(FOLLOW_term_in_expr215);
+                    term14=term();
+
+                    state._fsp--;
+
+                    adaptor.addChild(root_0, term14.getTree());
+
+                    }
+                    break;
+                case 3 :
+                    // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:33:11: STRING
+                    {
+                    root_0 = (Object)adaptor.nil();
+
+
+                    STRING15=(Token)match(input,STRING,FOLLOW_STRING_in_expr228); 
+                    STRING15_tree = 
+                    (Object)adaptor.create(STRING15)
+                    ;
+                    adaptor.addChild(root_0, STRING15_tree);
+
+
+                    }
+                    break;
+
+            }
+            retval.stop = input.LT(-1);
+
+
+            retval.tree = (Object)adaptor.rulePostProcessing(root_0);
+            adaptor.setTokenBoundaries(retval.tree, retval.start, retval.stop);
+
+        }
+        catch (RecognitionException re) {
+            reportError(re);
+            recover(input,re);
+    	retval.tree = (Object)adaptor.errorNode(input, retval.start, input.LT(-1), re);
+
+        }
+
+        finally {
+        	// do for sure before leaving
+        }
+        return retval;
+    }
+    // $ANTLR end "expr"
+
+
+    public static class term_return extends ParserRuleReturnScope {
+        Object tree;
+        public Object getTree() { return tree; }
+    };
+
+
+    // $ANTLR start "term"
+    // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:35:1: term : ( ( MULT | DIV ) ^ expr expr | factor );
+    public final EsperParser.term_return term() throws RecognitionException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.